usaspending_api/broker/models.py

```diff
@@ -37,3 +37,37 @@ class Meta:
         managed = True
         # Schema is set to `public` in migration `0003_deltatableloadversion`
         db_table = "delta_table_load_version"
+
+
+class LoadTracker(models.Model):
```

Comment:
I'd encourage you to combine these three tables into one if possible. Do the types and steps need to be tracked in separate tables?

Speaking from experience with `external_data_load_date` and `external_data_type`, I've always wished these two tables were originally created as a single table, so I don't have to join them every time I want to pull meaningful information from them.

Also, since you've taken the step to create a log of some of this history, what do you think about adding some extra info for incremental loads here like the Delta Load version or number of rows inserted/updated?

Reply:
> I'd encourage you to combine these three tables into one if possible. Do the types and steps need to be tracked in separate tables?

The tables are normalized right now. That's why I did what I did. Why should we favor denormalization over normalization for these table?

> Also, since you've taken the step to create a log of some of this history, what do you think about adding some extra info for incremental loads here like the Delta Load version or number of rows inserted/updated?

I made the very intentional decision to not because at this point in our pipeline Delta tables have no relation to anything being done. I believe it's going to better for a code base to not make a process that is unrelated to delta tables have to know about delta tables.

Reply:
If it's such a big deal, I could create a view on top of these tables that denormalizes it to save us the trouble. That would solve the problem you stated while still allowing us to follow best practices.

Reply:
> The tables are normalized right now. That's why I did what I did. Why should we favor denormalization over normalization for these tables?

I do believe normalization is an important aspect of database design, but in this case I can't see much value added. The tables `load_tracker_type` and `load_tracker_step` only consist of a single field (other than their id fields): `name`. Why are these lists of names important to normalize? To me normalization becomes more important if there are multiple data points that are being tracked in the side table that need to stay consistent with each other. (`address_id`, `address_street`, `address_zip`, etc) or even (`type_code` and `type_name` in the case of `award_category`). But for these single fields, it adds overhead for developers and the database (although database impact is negligible). 

As I mentioned above, the cost of normalization here is going to be that developers are going to have to juggle a few tables every time they want to pull information out of that table (unless they memorize all the ids).

Reply:
Here's a solution to your problem of cost to developers. I could create a view on top of these tables that denormalizes it to save us the trouble. That would solve the problem you stated while still allowing us to follow best practices.

Reply:
I'm good with that compromise 👍

Reply:
Done